plugins/de.fraunhofer.ipa.ros.xtext/src/de/fraunhofer/ipa/ros/Ros.xtext

```diff
@@ -6,221 +6,282 @@ import "http://www.ipa.fraunhofer.de/primitives" as primitives
-	'{'
-		('FromGitRepo' fromGitRepo=EString)?
-		('Dependencies' '{' dependency+=Dependency ( "," dependency+=Dependency)* '}' )?		
-		('Specs'  '{' spec+=SpecBase ( "," spec+=SpecBase)* '}')?
-		(artifact+=Artifact ( "," artifact+=Artifact)*)?
-	'}';
-	
+	name=RosNames':'
+	BEGIN
+		('fromGitRepo:' fromGitRepo=EString)?
+		('artifacts:' 
+			BEGIN 
+			artifact+=Artifact*
+			END
+		)?
+		('dependencies:' '[' dependency+=Dependency (',' dependency+=Dependency)* ']' )?
+	END;
+
 AmentPackage returns AmentPackage:
 	{AmentPackage}
 	'AmentPackage'
-	name=RosNames
-	'{'
-		('FromGitRepo' fromGitRepo=EString)?
-		('Dependencies' '{' dependency+=Dependency ( "," dependency+=Dependency)* '}' )?		
-		('Specs'  '{' spec+=SpecBase ( "," spec+=SpecBase)* '}')?
-		(artifact+=Artifact ( "," artifact+=Artifact)*)?
-	'}';
+	name=RosNames':'
+	BEGIN
+		('fromGitRepo:' fromGitRepo=EString)?
+		('artifacts:' 
+			BEGIN 
+			artifact+=Artifact*
+			END
+		)?
+		('dependencies:' '[' dependency+=Dependency (',' dependency+=Dependency)* ']' )?
+	END;
+
+///////////////////
+//ARTIFACT AND NODE
+///////////////////
+Artifact returns Artifact:
+	{Artifact}
+		name=RosNames':'
+		BEGIN
+		(node=Node)?
+		END;
 
-ServiceSpec returns ServiceSpec:
-	{ServiceSpec}
-	'ServiceSpec'
-	name=EString
-	'{'
-		('request' request=MessageDefinition)?
-		('response' response=MessageDefinition)?
-	'}';
+Node returns Node:
+	'node:' name=RosNames
+	 BEGIN
+		(
+		('publishers:' 
+			BEGIN 
+			publisher+=Publisher*
+			END
+		)|
+		('subscribers:' 
+			BEGIN 
+			subscriber+=Subscriber*
+			END
+		)|
+		('serviceserver:' 
+			BEGIN 
+			serviceserver+=ServiceServer*
+			END
+		)|
+		('serviceclient:' 
+			BEGIN 
+			serviceclient+=ServiceClient*
+			END
+		)|
+		('actionserver:' 
+			BEGIN 
+			actionserver+=ActionServer*
+			END
+		)|
+		('actionclient:' 
+			BEGIN 
+			actionclient+=ActionClient*
+			END
+		)|
+		('parameters:'
+			BEGIN
+			parameter+=Parameter*
+			END
+		)
+		)*END
+	;
+
+///////////////////
+//OBJECTS/SPECIFICATIONS
+///////////////////
+SpecBase returns SpecBase:
+	TopicSpec | ServiceSpec | ActionSpec;
 
 TopicSpec returns TopicSpec:
 	{TopicSpec}
-	'TopicSpec'
-	name=(EString|'Header'|'String')
-	'{'
-		('message' message=MessageDefinition)?
-	'}';
+	'msg:'name=(EString|'Header'|'String')
+	BEGIN 
+		'message:' (BEGIN message=MessageDefinition END)?
+	END
+	;
 
+ServiceSpec returns ServiceSpec:
+	{ServiceSpec}
+	'srv:'name=EString
+	BEGIN
+		'request:' (BEGIN request=MessageDefinition END)?
+		'response:' (BEGIN response=MessageDefinition END)?
+	END;
+	
 ActionSpec returns ActionSpec:
 	{ActionSpec}
-	'ActionSpec'
-	name=EString
-	'{'
-		('goal' goal=MessageDefinition)?
-		('result' result=MessageDefinition)?
-		('feedback' feedback=MessageDefinition)?
-	'}';
+	'action:'name=EString
+	BEGIN
+		'goal:' (BEGIN goal=MessageDefinition END)?
+		'result:' (BEGIN result=MessageDefinition END)?
+		'feedback:' (BEGIN feedback=MessageDefinition END)?
+	END;
```

Comment:
Seems to have the same YAML issues. Probbaly add name tag as well.

Reply:
You are right, I made already the fix for other branch: https://github.com/ipa320/ros-model/blob/7d09541be7f5339c09fb79ccecfe32abf21ec2ed/plugins/de.fraunhofer.ipa.ros.xtext/src/de/fraunhofer/ipa/ros/Ros.xtext#L121-L144

I think we came up to exactly the same solution :) 

This is part of https://github.com/ipa-nhg/ros-model/tree/YamlMigrationSystem ‼️ that branch is not passing the tests